libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/detail/search.hpp

```diff
@@ -0,0 +1,153 @@
+//  Copyright (c) 2020 ETH Zurich
+//  Copyright (c) 2014 Grant Mercer
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//  This file is based on the following cppreference possible implementation:
+//  https://en.cppreference.com/w/cpp/algorithm/ranges/search
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/algorithms/traits/projected.hpp>
+#include <hpx/functional/invoke.hpp>
+#include <hpx/execution/algorithms/detail/predicates.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+#include <hpx/parallel/util/loop.hpp>
+#include <hpx/parallel/util/partitioner.hpp>
+
+
+#include <cstddef>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx { namespace ranges { namespace detail {
```

Comment:
Following @hkaiser's comment about moving this file, I'd also move this back to `hpx::parallel::v1::detail` even though it does also deal with ranges. The implementation is used for both the ranges and non-ranges versions though and if nothing else it'll stay more consistent with what we've done so far. Or do we want to start moving the algorithm *implementations* into `hpx::ranges`?

Reply:
I'll let @hkaiser reply, I don't really mind

Reply:
There is no reason at this point to move this away from `parallelism::v1`, I think.